### PR TITLE
Fix rest_server main.go errors, go mod tidy

### PR DESCRIPTION
**Pull request by kevya-google**
Fix rest_server slog errors from https://github.com/abcxyz/abc/pull/157.

Can merge to https://github.com/abcxyz/abc/tree/kevya/template-subfolder first or main if https://github.com/abcxyz/abc/pull/171 merges first.

### CODE REVIEW

File: t/rest_server/contents/main.go [hunk 1]
```diff
@@ -94,8 +95,8 @@ func main() {
 
 	flag.Parse()
 	if err := realMain(logging.WithLogger(ctx, logger)); err != nil {
-		done() // deferred function calls won't execute due to logger.Fatal() never returning
-		logger.Fatal(err)
```
Comment (drevell) [inline t/rest_server/contents/main.go:98]: Shouldn't we log this error?

**Review by drevell — CHANGES_REQUESTED** [review]

File: t/rest_server/contents/main.go (hunk 1 above)
Comment (kevya-google) [reply t/rest_server/contents/main.go:98]: github.com/abcxyz/pkg/logging's logger doesn't have a Fatal. What should I do?

Comment (drevell) [reply t/rest_server/contents/main.go:98]: Well, the behavior of Fatal was to log then exit, so I think you can log at the most severe level available. And there's already a call to Exit() below.

Comment (kevya-google) [reply t/rest_server/contents/main.go:98]: Ok, I was going to do logging.LevelEmergency but I think LevelError sounded more appropriate.

Comment (drevell) [reply t/rest_server/contents/main.go:98]: I think logger.ErrorContext() is more conventional than logger.Log(..., logging.LevelError)

Comment (kevya-google) [reply t/rest_server/contents/main.go:98]: Got it, thank you!

**Review by drevell — APPROVED** [review]

**Pull request merged**